pandas/io/excel/_util.py

```diff
@@ -145,7 +161,29 @@ def _range2cols(areas: str) -> list[int]:
     return cols
 
 
-def maybe_convert_usecols(usecols):
+@overload
+def maybe_convert_usecols(usecols: str | list[int]) -> list[int]:
+    ...
+
+
+@overload
+def maybe_convert_usecols(usecols: list[str]) -> list[str]:
+    ...
+
+
+@overload
+def maybe_convert_usecols(usecols: usecols_func) -> usecols_func:
+    ...
+
+
+@overload
+def maybe_convert_usecols(usecols: None) -> None:
+    ...
+
+
+def maybe_convert_usecols(
+    usecols: str | list[int] | list[str] | usecols_func | None,
```

Comment:
prob can share this with the parsers (followon, maybe put in _typing.py)?

Reply:
The configs there are a bit different unfortunately. Don't think we can share efficiently